File: python/meteroid/models/subscription_update_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .payment_methods_config import PaymentMethodsConfig


@dataclasses.dataclass
class SubscriptionUpdateRequest(BaseModel):
    auto_advance_invoices: t.Optional[bool] = None
    """If false, invoices will stay in Draft until manually reviewed and finalized."""

    charge_automatically: t.Optional[bool] = None
    """Automatically try to charge the customer's configured payment method on finalize."""

    custom_properties: t.Optional[t.Dict[str, t.Any]] = None
    """Partial update of custom property values (merge; send a key with `null` to remove it).
    Validated against the tenant's `SUBSCRIPTION` property definitions. Omit to leave unchanged."""

    invoice_memo: t.Optional[str] = None
    """Default memo for invoices"""

    net_terms: t.Optional[int] = None
    """Payment terms in days (0 = due on issue)"""

    payment_methods_config: t.Optional[PaymentMethodsConfig] = None

    purchase_order: t.Optional[str] = None
    """Purchase order number"""

File: go/subscription_update_request.go
// this file is @generated
package meteroid

type SubscriptionUpdateRequest struct {
	// If false, invoices will stay in Draft until manually reviewed and finalized.
	AutoAdvanceInvoices *bool `json:"auto_advance_invoices,omitempty"`

	// Automatically try to charge the customer's configured payment method on finalize.
	ChargeAutomatically *bool `json:"charge_automatically,omitempty"`

	// Partial update of custom property values (merge; send a key with `null` to remove it).
	// Validated against the tenant's `SUBSCRIPTION` property definitions. Omit to leave unchanged.
	CustomProperties map[string]any `json:"custom_properties,omitempty"`

	// Default memo for invoices
	InvoiceMemo *string `json:"invoice_memo,omitempty"`

	// Payment terms in days (0 = due on issue)
	NetTerms *int32 `json:"net_terms,omitempty"`

	PaymentMethodsConfig *PaymentMethodsConfig `json:"payment_methods_config,omitempty"`

	// Purchase order number
	PurchaseOrder *string `json:"purchase_order,omitempty"`
}
